components/browser/session/src/main/java/mozilla/components/browser/session/SelectionAwareSessionObserver.kt

```diff
@@ -44,6 +44,19 @@ abstract class SelectionAwareSessionObserver(
         activeSession?.register(this)
     }
 
+    /**
+     * Starts observing changes to the session matching the [sessionId].
+     *
+     * @param sessionId the session ID to observe.
+     */
+    fun observeFromId(sessionId: String?) {
+        sessionId?.let {
```

Comment:
This is definitely correct, but do you mind going back to the old version of the code? I find it reads better.